packages/app/src/cli/services/deploy/upload.ts

```diff
@@ -169,6 +170,83 @@ interface UploadFunctionExtensionsOptions {
   identifiers: Identifiers
 }
 
+interface UploadFunctionExtensionOptions {
+  apiKey: string
+  identifier?: string
+  token: string
+}
+
+/**
+ * Uploads a bundle.
+ * @param options - The upload options
+ */
+ export async function uploadFunctionExtensionsWithEF(
+  extensions: FunctionExtension[],
+  options: UploadFunctionExtensionsOptions,
+): Promise<UploadExtensionValidationError[]> {
+  const deploymentUUID = randomUUID()
+  // const signedURL = await getUIExtensionUploadURL(options.identifiers.app, deploymentUUID)
+
+  // Must upload Wasm blobs in sequence. Otherwise we run into "429 too many requests".
+  let signedURLs: {
+    [localIdentifier: string]: string
+  } = {}
+
+  for (const extension of extensions) {
+    const url = await uploadWasmBlob(extension, options.identifiers.app, options.token)
+    signedURLs[extension.localIdentifier] = url
+    await uploadFunctionExtension(extension, url, {
+      apiKey: options.identifiers.app,
+      token: options.token,
+      identifier: options.identifiers.extensions[extension.localIdentifier],
+    })
+  }
+
+  const formattedExtensions: ExtensionSettings[] = await Promise.all(
+    extensions.map(async (extension) => {
+      return {
+        uuid: options.identifiers.extensions[extension.localIdentifier] || "error",
+        config: JSON.stringify(await functionConfiguration(extension, signedURLs[extension.localIdentifier])),
+        context: '',
+      }
+    })
+  )
+
+  // output.info("formattedExtensions " + JSON.stringify(formattedExtensions))
+
+  // const formData = http.formData()
+  // const buffer = readSync("./")
+  // formData.append('my_upload', buffer)
+  // await http.fetch(signedURL, {
+  //   method: 'put',
+  //   body: buffer,
+  //   headers: formData.getHeaders(),
+  // })
+
+  const variables: CreateDeploymentVariables = {
+    apiKey: options.identifiers.app,
+    uuid: deploymentUUID,
+    bundleUrl: Object.values(signedURLs)[0] || "", // we don't use this, but it must be a valid URL for deploy to work.
+    extensions: formattedExtensions,
+  }
+
+  const mutation = CreateDeployment
+  const result: CreateDeploymentSchema = await partnersRequest(mutation, options.token, variables)
```

Comment:
We are already doing a `CreateDeployment` query for UI extensions and Theme extensions, we should try to just add these new `fromattedExtensions` there. It's on line 118 of this same file. And the extensions object is created in `deploy.ts` line 58: you can see we create an array of extensions (we should append there the `formattedExtensions) and then we pass it to upload.

Let me know if you need help with this, we can pair!

Reply:
This is just here so I can see the diff, there's no intent to merge.

That said, we will happily take you up on that pairing offer once we get into build 😄

Reply:
sure, just ping me on slack whenever you are ready to move forward with this :)